src/spec.c

```diff
@@ -1146,7 +1146,7 @@ static void IndexSpec_ScanCallback(RedisModuleCtx *ctx, RedisModuleString *keyna
   }
   dictReleaseIterator(iter);
 
-  Document_Free(&doc);
+  Document_Free(&aCtx->doc);
```

Comment:
The aCtx here is already freed on AddDocumentCtx_Submit, we can add a force flag to Document_Free?